source/extensions/quic_listeners/quiche/envoy_quic_alarm.cc

```diff
@@ -12,7 +12,15 @@ void EnvoyQuicAlarm::CancelImpl() { timer_->disableTimer(); }
 
 void EnvoyQuicAlarm::SetImpl() {
   // TODO(#7170) switch to use microseconds if it is supported.
-  timer_->enableTimer(std::chrono::milliseconds(getDurationBeforeDeadline().ToMilliseconds()));
+  quic::QuicTime::Delta duration = getDurationBeforeDeadline();
+  // Round up the duration so that any duration < 1ms will not be
+  // triggered within current event loop.
```

Comment:
What exactly is the problem? Is it the alarm fires within this function? or within the current event loop?

Reply:
The problem is when duration is 0 the alarm gets fired within current event loop. If the alarm firing doesn't make any forward progress in QUICHE, we would have infinite loop.